onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -65,22 +150,32 @@ def augment_graph(self):
         tensors_to_calibrate = set()
 
         for node in model.graph.node:
-            should_be_calibrate = ((node.op_type in self.calibrate_op_types) and
-                                   (node.name not in self.black_nodes)) or (node.name in self.white_nodes)
+            if augment_all_ops:
+                should_be_calibrate = True
+            else:
+                should_be_calibrate = ((node.op_type in self.calibrate_op_types) and
+                                       (node.name not in self.black_nodes)) or (node.name in self.white_nodes)
             if should_be_calibrate:
                 for tensor_name in itertools.chain(node.input, node.output):
                     if tensor_name in value_infos.keys():
                         vi = value_infos[tensor_name]
                         if vi.type.HasField('tensor_type') and vi.type.tensor_type.elem_type == TensorProto.FLOAT and (
                                 tensor_name not in model.graph.initializer):
                             tensors_to_calibrate.add(tensor_name)
+        
+        # If augmenting all ops, it's possible that some nodes' input value are 0.
```

Comment:
> input value are 0. [](start = 63, length = 19)

do you mean a 1d tensor?

Reply:
1d tensor has 0 dimension on axis.
  The code is in reduction_ops.cc:

Reply:
if (axes.size() > 0) {
      prepare_reduce_metadata.output_dims = input_dims;
      for (auto axis : axes) {
        axis = HandleNegativeAxis(axis, rank);
        ORT_ENFORCE(input_dims[axis] != 0,
                      "Can't reduce on dim with value of 0 if 'keepdims' is false. "
                     "Invalid output shape would be produced. input_shape:",
                    input_shape);
        prepare_reduce_metadata.output_dims[axis] = 1;
        reduced[axis] = true;
      }
    } else {
      // no axes provided (i.e.) default axes  => reduce on all dims
      for (auto dim : input_dims) {
        ORT_ENFORCE(keepdims || dim != 0,
                    "Can't reduce on dim with value of 0 if 'keepdims' is false. "
                    "Invalid output shape would be produced. input_shape:",
                    input_shape);
        prepare_reduce_metadata.output_dims.push_back(dim == 0 ? 0 : 1);
      }
    }